bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/tools/DelayedBatchProcessing.java

```diff
@@ -56,18 +57,19 @@ public DelayedBatchProcessing(int delay, Consumer<List<T>> consumer, ScheduledEx
     }
 
     /**
-     * Add new object to the batch process list. If the list was empty, the delay timer
-     * is armed and all successive objects are accumulated from here on.
+     * Add new object to the batch process list. Every time a new object is received,
+     * the delay timer is rescheduled.
      *
      * @param t An object
      */
     @Override
-    public void accept(T t) {
+    synchronized public void accept(T t) {
         queue.add(t);
-        final ScheduledFuture<?> scheduledFuture = this.future;
-        if (scheduledFuture == null || scheduledFuture.isDone()) {
-            this.future = executor.schedule(this::run, delay, TimeUnit.MILLISECONDS);
+        final ScheduledFuture<?> scheduledFuture = this.futureRef.get();
+        if (scheduledFuture != null && !scheduledFuture.isDone()) {
+            cancel(futureRef.getAndSet(null));
         }
```

Comment:
The cancel is already handled in the next statement. So you can remove this logic.
```suggestion
```